code/src/layouts/header/components/header-toolbar/HeaderToolbar.tsx

```diff
@@ -71,10 +74,37 @@ const HeaderToolbar = () => {
     openDrawer(<CartDrawer />);
   };
 
-  const badgeContentTypography = (
-    <AppTypography variant="caption-small">{itemsInCartCount}</AppTypography>
+  useEffect(() => {
+    if (user?.id && isAuthenticated) {
+      fetch(user.id);
+    }
+  }, [user?.id]);
+
+  const cartItemsCount = data?.items.length;
```

Comment:
Now the items counting isn't correct, because you use items.length but not quantity
I offer to change in this way
`const cartItemsCount = data?.items.reduce((total, item) => total + item.quantity, 0)`
It is how cart.data looks in console
`
{items: Array(1), totalPrice: 200.9}
items
: 
Array(1)
0
: 
calculatedPrice
: 
200.9
image
: 
"https://j65jb0fdkxuua0go.public.blob.vercel-storage.com/phone_2-tTDYhyoyqsEkwPzySFdXflYCe7TkUb.jpg"
name
: 
"Iphone"
productId
: 
"8efbee82-8a0c-407a-a4c0-16bbad40a23e"
productPrice
: 
100.45
quantity
: 
2
[[Prototype]]
: 
Object
length
: 
1
[[Prototype]]
: 
Array(0)
totalPrice
: 
200.9`

Reply:
Thank you for suggestion. I initially thought to not include quantity and count item as one always, even if its quantity is bigger than one (somehow it was logical to me). But your suggestion is the right one, I checked other e-commerce websites, and they have it the same way as you proposed.